dashboard/app/controllers/registrations_controller.rb

```diff
@@ -209,12 +209,18 @@ def begin_sign_up_params
     params.require(:user).permit(:email, :password, :password_confirmation)
   end
 
-  # Set age for the current user if empty - skips CSRF verification because this can be called
+  # Set age, us_state and gender for the current user if empty - skips CSRF verification because this can be called
   # from cached pages which will not populate the CSRF token. This also skips lockout policy
   # checks since those depend on the age being set.
-  def set_age
+  def set_student_information
     return head(:forbidden) unless current_user
-    current_user.update(age: params[:user][:age]) if current_user.age.blank?
+    student_information = {}
+
+    student_information[:age] = params[:user][:age] if current_user.age.blank?
+    student_information[:us_state] = params[:user][:us_state] if current_user.us_state.blank?
+    student_information[:gender_student_input] = params[:user][:gender_student_input] if current_user.gender.blank?
```

Comment:
Why do we only set these if the existing value is blank? Is there a reason we don't want to let them overwrite existing values?

Reply:
No reason i can think of, I was following the previous method which only set the age if the exisitng value was blank